### ISSUE TITLE
403 error when making API req from directus flows

### ISSUE THREAD

**Issue by JackMullinkosson** [directus/directus#21877]
### Describe the Bug

From the browser, curl, and a Next.js app using axios I receive a 200 response with the relevant data, but when I make the same call in Directus flows GET request, I get a 403 Forbidden response. 

### To Reproduce

1. Create an [eventbrite](https://www.eventbrite.com/) account
2. Create an [API key](https://www.eventbrite.com/account-settings/apps)
3. Copy the Private Token from that same page ^
4. Create a directus flow that makes a get request to https://www.eventbriteapi.com/v3/users/me/orders/?token=YOURTOKEN with no body and no headers.

### Directus Version

v10.10.4

### Hosting Strategy

Directus Cloud

**Comment by br41nslug** [comment]
What is the returned error from evenbtbrite? A quick test without api key returns the same 401 error both from the browser and in the flow.

**Comment by JackMullinkosson** [comment]
@br41nslug here is the full error:
`{
  "status": 403,
  "statusText": "Forbidden",
  "headers": {
    "server": "CloudFront",
    "date": "Fri, 15 Mar 2024 14:40:26 GMT",
    "content-type": "text/html",
    "content-length": "915",
    "x-amz-cf-id": "PAAD1zR63zerB6bniQaMkcyG0C3PqD1dhzpHdru3UV0uVD2ARibwxg==",
    "x-cache": "Error from cloudfront",
    "via": "1.1 25e2963eb5d8a7965bc8b98c455aab48.cloudfront.net (CloudFront)",
    "x-amz-cf-pop": "IAD66-C2"
  },
  "data": "<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 4.01 Transitional//EN\" \"http://www.w3.org/TR/html4/loose.dtd\">\n<HTML><HEAD><META HTTP-EQUIV=\"Content-Type\" CONTENT=\"text/html; charset=iso-8859-1\">\n<TITLE>ERROR: The request could not be satisfied</TITLE>\n</HEAD><BODY>\n<H1>403 ERROR</H1>\n<H2>The request could not be satisfied.</H2>\n<HR noshade size=\"1px\">\nBad request.\nWe can't connect to the server for this app or website at this time. There might be too much traffic or a configuration error. Try again later, or contact the app or website owner.\n<BR clear=\"all\">\nIf you provide content to customers through CloudFront, you can find steps to troubleshoot and help prevent this error by reviewing the CloudFront documentation.\n<BR clear=\"all\">\n<HR noshade size=\"1px\">\n<PRE>\nGenerated by cloudfront (CloudFront)\nRequest ID: PAAD1zR63zerB6bniQaMkcyG0C3PqD1dhzpHdru3UV0uVD2ARibwxg==\n</PRE>\n<ADDRESS>\n</ADDRESS>\n</BODY></HTML>"
}`

**Comment by JackMullinkosson** [comment]
@br41nslug Does the Needs Research tag indicate you need something from my end?

**Comment by rafmsou** [comment]
I'm having a similar issue on Directus Cloud.

I have a AWS Cloudfront distribution to cache images from `/assets` and it used to work perfectly until a few days ago. Suddenly some images started to return an error when missed from the cache:
```
502 ERROR
The request could not be satisfied.
CloudFront wasn't able to connect to the origin. We can't connect to the server for this app or website at this time. There might be too much traffic or a configuration error. Try again later, or contact the app or website owner.
If you provide content to customers through CloudFront, you can find steps to troubleshoot and help prevent this error by reviewing the CloudFront documentation.
Generated by cloudfront (CloudFront)
Request ID: 5WtavyGLGaYLQ-kfWvuOX3CFl1Tx67MFgt_6Vc1MFhq5FK1sQK8Bbw==
``` 
I reviewed all settings related to the TLS version and made sure I have everything on the latest and but nothing changed.
I suspect this started after the TLS1.3 update that happened on Directus Cloud.

**Comment by JackMullinkosson** [comment]
@rafmsou Agreed, it is likely related to the TLS update because Eventbrite (along with many other API's) will not accept requests from 1.3.

**Comment by br41nslug** [comment]
> TLS update because Eventbrite (along with many other API's) will not accept requests from 1.3.

If so then eventbrite will really need an update, TLS 1.3 has been out since 2018 and best security practices are to only accept this protocol to prevent protocol downgrading attacks

**Comment by JackMullinkosson** [comment]
Hi @br41nslug I just retested and am still getting the same result. Works in browser but same 403 forbidden error when sent from Directus flow.

**Comment by br41nslug** [comment]
Have you contacted Eventbrite support for this? `CloudFront wasn't able to connect to the origin.` is an error thats coming from their cloud setup perhaps they shine some light on why this is erroring. We havent changed anything in relation to outgoing requests in the core application or on cloud.

**Comment by JackMullinkosson** [comment]
I did contact Eventbrite support to explain the issue. You tested it on an instance of directus cloud? Would you mind sending screenshots of what the request and response looked like, or even better a link to the flow?

**Comment by licitdev** [comment]
@JackMullinkosson I followed your reproduction steps on a Cloud project and the query was successful. I was also able to successfully run the same request from flows locally. Kindly provide additional details of your flow as I'm unable to reproduce the 403 error.

<img width="776" alt="image" src="https://github.com/directus/directus/assets/26413686/7fc1a46d-9e09-4ccb-87ac-b0fb7408c259">


<img width="698" alt="image" src="https://github.com/directus/directus/assets/26413686/5b492c15-a2c1-4c23-a7c2-a9734321b01c">

<details>
  <summary>Request response from flow logs</summary>

```json
{
  "status": 200,
  "statusText": "OK",
  "headers": {
    "content-type": "application/json",
    "content-length": "594",
    "connection": "close",
    "date": "Thu, 28 Mar 2024 16:58:33 GMT",
    "server": "nginx",
    "x-ua-compatible": "IE=edge",
    "referrer-policy": "strict-origin-when-cross-origin",
    "allow": "GET, HEAD, OPTIONS",
    "x-content-type-options": "nosniff",
    "access-control-max-age": "600",
    "expires": "0",
    "vary": "Accept, Accept-Encoding",
    "x-xss-protection": "1; mode=block",
    "strict-transport-security": "max-age=63072000; includeSubDomains; preload",
    "x-rate-limit": "token:REDACTED 3/2000 reset=3513s, key:REDACTED 3/2000 reset=3513s",
    "pragma": "no-cache",
    "cache-control": "no-cache, no-store, must-revalidate, max-age=0",
    "x-frame-options": "SAMEORIGIN",
    "access-control-allow-origin": "*",
    "access-control-allow-headers": "Authorization, Content-Type, Eventbrite-API-Waypoint-Token, Eventbrite-API-Deprecated-Features",
    "p3p": "CP=\"NOI ADM DEV PSAi COM NAV OUR OTRo STP IND DEM\"",
    "x-permitted-cross-domain-policies": "none",
    "set-cookie": [
      "stableId=REDACTED; Domain=.eventbriteapi.com; expires=Fri, 28-Mar-2025 16:58:33 GMT; Path=/; secure",
      "mgrefby=; Domain=.eventbriteapi.com; expires=Fri, 28-Mar-2025 16:58:33 GMT; httponly; Max-Age=31536000; Path=/; secure",
      "G=REDACTED; Domain=.eventbriteapi.com; expires=Fri, 28-Mar-2025 16:58:33 GMT; httponly; Path=/; secure",
      "ebEventToTrack=; Domain=.eventbrite.com; expires=Sat, 27-Apr-2024 16:58:33 GMT; httponly; Path=/; secure",
      "SS=REDACTED; Domain=.eventbriteapi.com; httponly; Path=/; secure",
      "eblang=lo%3Den_US%26la%3Den-us; Domain=.eventbriteapi.com; expires=Fri, 28-Mar-2025 16:58:33 GMT; httponly; Path=/; secure",
      "AN=; Domain=.eventbrite.com; expires=Sat, 27-Apr-2024 16:58:33 GMT; httponly; Path=/; secure",
      "AS=REDACTED; Domain=.eventbriteapi.com; httponly; Path=/; secure",
      "mgref=typeins; Domain=.eventbriteapi.com; expires=Fri, 28-Mar-2025 16:58:33 GMT; httponly; Max-Age=31536000; Path=/; secure",
      "SP=REDACTED; Domain=.eventbriteapi.com; httponly; Path=/; secure"
    ],
    "x-cache": "Miss from cloudfront",
    "via": "1.1 REDACTED.cloudfront.net (CloudFront)",
    "x-amz-cf-pop": "IAD66-C2",
    "x-amz-cf-id": "REDACTED"
  },
  "data": {
    "pagination": {
      "object_count": 5,
      "page_number": 1,
      "page_size": 50,
      "page_count": 1,
      "has_more_items": false
    },
    "orders": [
      {
        "costs": {
          "base_price": {
            "display": "$0.00",
            "currency": "SGD",
            "value": 0,
            "major_value": "0.00"
          },
          "eventbrite_fee": {
            "display": "$0.00",
            "currency": "SGD",
            "value": 0,
            "major_value": "0.00"
          },
          "gross": {
            "display": "$0.00",
            "currency": "SGD",
            "value": 0,
            "major_value": "0.00"
          },
          "payment_fee": {
            "display": "$0.00",
            "currency": "SGD",
            "value": 0,
            "major_value": "0.00"
          },
          "tax": {
            "display": "$0.00",
            "currency": "SGD",
            "value": 0,
            "major_value": "0.00"
          }
        },
        "resource_uri": "https://www.eventbriteapi.com/v3/orders/REDACTED/",
        "id": "REDACTED",
        "changed": "2019-07-14T14:53:45Z",
        "created": "2019-07-14T14:53:44Z",
        "name": "REDACTED",
        "first_name": "REDACTED",
        "last_name": "REDACTED",
        "email": "REDACTED",
        "status": "placed",
        "time_remaining": null,
        "is_pending_order_form": false,
        "event_id": "REDACTED"
      }
    ]
  }
}
```
</details>

**Comment by rafmsou** [comment]
Updating here.

I asked the Cloud support to revert the TLS upgrade/changes and it worked after that. So that's where the problem is.

The thing is Cloudfront seems be to compatible with TLS1.3 a long time ago and we can see it here: 
https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/secure-connections-supported-viewer-protocols-ciphers.html

I also tried to troubleshoot following [this guide](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/http-502-bad-gateway.html) and everything seems ok as far as I could check, so it was supposed to be working.

I believe whoever is using a Cloudfront distribution to cache Directus data will have the same problem, can anyone try that?

I even tried creating another distribution from scratch reviewing all the steps but ended up having the same issue.

**Comment by licitdev** [comment]
@rafmsou Yes there's support for TLS1.3 between AWS CloudFront and viewers, but it does not support TLS1.3 when it connects to an origin server.
https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/secure-connections-supported-ciphers-cloudfront-to-origin.html
